siad/daemon.go

```diff
@@ -34,80 +37,127 @@ func processNetAddr(addr string) string {
 
 // processConfig checks the configuration values and performs cleanup on
 // incorrect-but-allowed values.
-func processConfig(config Config) Config {
+func processConfig(config Config) (Config, error) {
 	config.Siad.APIaddr = processNetAddr(config.Siad.APIaddr)
 	config.Siad.RPCaddr = processNetAddr(config.Siad.RPCaddr)
 	config.Siad.HostAddr = processNetAddr(config.Siad.HostAddr)
-	return config
+	config.Siad.Modules = strings.ToLower(config.Siad.Modules)
+	validModules := "cghmrtwe"
+	invalidModules := config.Siad.Modules
+	for _, m := range validModules {
+		invalidModules = strings.Replace(invalidModules, string(m), "", 1)
+	}
+	if len(invalidModules) > 0 {
+		return Config{}, errors.New("Unable to parse --modules flag, unrecognized modules: " + invalidModules)
+	}
```

Comment:
I think this will break if you supplied the same letter twice. This is kind of a hairy problem. Hmm